src/components/Reports/GoalCalculator/GoalCalculator.tsx

```diff
@@ -66,13 +114,32 @@ export const GoalCalculator: React.FC<GoalCalculatorProps> = ({
     selectedCategoryId,
     selectedStepId,
     currentStep,
+    isRightOpen,
+    isDrawerOpen,
     handleCategoryChange,
     handleStepChange,
+    toggleRightPanel,
+    toggleDrawer,
+    setDrawerOpen,
   } = useGoalCalculator();
   const { t } = useTranslation();
+  const iconPanelWidth = theme.spacing(5);
+
+  const handleCategoryIconClick = (categoryId: typeof selectedCategoryId) => {
```

Comment:
In a follow-up PR at some point, I'd love to see this logic moved to the context. But it's not crucial and can wait.